### PR TITLE
CAN 2.0B Driver changes

### PR DESCRIPTION
**Pull request by JasonBrave**
Update CAN driver only, update SID will be done in a separate PR

### CODE REVIEW

File: pic18f26k83/pic18f26k83_can.c [hunk 1]
```diff
@@ -151,7 +158,12 @@ void can_handle_interrupt() {
     } else if (PIR5bits.RXB1IF) {
 
         can_msg_t rcvd_msg;
-        rcvd_msg.sid = (((uint16_t)RXB1SIDH) << 3) | (RXB1SIDL >> 5);
+        rcvd_msg.sid = (uint32_t)RXB1SIDH << 21;
+        rcvd_msg.sid |= (((uint32_t)RXB1SIDL >> 5) & 0x7) << 18;
+        rcvd_msg.sid |= ((uint32_t)RXB1SIDL & 0x3) << 16;
+        rcvd_msg.sid |= (uint32_t)RXB1EIDH << 8;
+        rcvd_msg.sid |= RXB1EIDL;
```
Comment (celery6) [inline pic18f26k83/pic18f26k83_can.c:165]: add more comments so we know what this is doing at a glance?

**Review by celery6 — COMMENTED** [review]
can u add more comments for the bitshifting? make it more maintainable if someone else needs to debug it or anything.

File: pic18f26k83/pic18f26k83_can.c (hunk 1 above)
Comment (JasonBrave) [reply pic18f26k83/pic18f26k83_can.c:165]: I can put a link to datasheet if you want?

Comment (celery6) [reply pic18f26k83/pic18f26k83_can.c:165]: i was thinking smth like "store the [N:n] bits from rxb1eid into sid" or something like that to describe the bit shifting. unless its really easy to interpret these 5 bitwise ops and im just skill issuing?

Comment (Rio6) [reply pic18f26k83/pic18f26k83_can.c:165]: adding datasheet page number is probably good enough

**Review by AshTheEngiqueer — APPROVED** [review]
LGTM :)

**Review by celery6 — APPROVED** [review]

**Pull request merged**